File: Lessons/04_CountingElements_FrogRiverOne.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(x, a)
    # write your code in Ruby 2.2
    hash = {} 
    cnt = 0
    a.length.times do |i|
      cnt += 1 if hash.has_key?(a[i]) == false 
      return i if cnt == x 
      hash[a[i]] = true 
    end
    return -1
end
  

File: Python/04_CountingElements_FrogRiverOne.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X, A):
    # write your code in Python 3.6
    dic = {}
    for i in range(len(A)):
        dic[A[i]] = True 
        if len(dic) >= X: return i
    
    return -1
# end of func